drivers/scheduler/k8s/k8s.go

```diff
@@ -5055,35 +5055,56 @@ func (k *K8s) CreateCsiSnapsForVolumes(ctx *scheduler.Context, snapClass string)
 	return volSnapMap, nil
 }
 
-// CreateCsiSnapsAndValidate create CSI snapshot for Pure FA volumes and restore them to new PVCs, and validate the content
+// CreateCsiSnapsAndValidate create CSI snapshot for volumes and restore them to new PVCs, and validate the content
 // (Testrail cases C58775, 58091)
-func (k *K8s) CreateCsiSnapsAndValidate(ctx *scheduler.Context, snapClass string) error {
+func (k *K8s) CreateCsiSnapsAndValidate(ctx *scheduler.Context, request scheduler.CSISnapshotRequest) error {
 	// This test will validate the content of the snapshot as opposed to just verify creation of snapshot.
-	timestamp := strings.Split(snapClass, "-")[0]
-	namespace := "fada-basic-snapshot-and-restore-test-" + timestamp
+	timestamp := request.Timestamp
+	namespace := request.Namespace + timestamp
 	clientset, err := getKubeClient("")
 	if err != nil {
-		logrus.Errorf("Failed to get kube client for FADA snapshot and restore test: %s", err)
+		logrus.Errorf("Failed to get kube client for snapshot and restore test: %s", err)
 		return err
 	}
+	storageClassName := request.StorageclassName
+	// for FADA test, find a FA SC on our own that has pure_block param, if no FA SC, skip the test
+	if request.FADA {
```

Comment:
There may be multiple FADA storage classes installed if we run multiple apps: are we not able to use the PVC to get the PV and then use the storage class name from there?

Reply:
I don't understand why you need `request.FADA`? Why can't we just use `request.storageClassName` you pass in this function?

Reply:
@Pure-AdamuKaapan the test is not associated with any app. Its just like our PSO app, we are using a single pod for this test. Because other apps may have data written to it already and I want this volume to contain only the dirty data for ease of comparison. So it just spins up a simple pod for verification purpose. 
@dsupure since we might be supporting FB snapshot/cloning later but now we aren't. But currently this test is also triggered for FB causing FBDA torpedo failure yesterday. with this parameter, we go in to find a FADA SC, and if it doesn't find it, it will skip the test. But if other people want to just try a CSI clone, they can still call this function, and set FADA to false and still be able to use it